lib/analyzer/image-inspector.ts

```diff
@@ -1,23 +1,22 @@
 import { Docker, DockerOptions } from "../docker";
 import { DockerInspectOutput } from "./types";
 
-export { detect };
+export { detect, pullIfNotLocal };
 
 async function detect(
   targetImage: string,
   options?: DockerOptions,
 ): Promise<DockerInspectOutput> {
+  const docker = new Docker(targetImage, options);
+  const info = await docker.inspectImage(targetImage);
+  return JSON.parse(info.stdout)[0];
+}
+
+async function pullIfNotLocal(targetImage: string, options?: DockerOptions) {
+  const docker = new Docker(targetImage);
   try {
-    const info = await new Docker(targetImage, options).inspectImage(
-      targetImage,
-    );
-    return JSON.parse(info.stdout)[0];
-  } catch (error) {
-    if (error.stderr.includes("No such object")) {
-      throw new Error(
-        `Docker error: image was not found locally: ${targetImage}`,
-      );
-    }
-    throw new Error(`Docker error: ${error.stderr}`);
+    await docker.inspectImage(targetImage);
+  } catch (err) {
+    await docker.pull(targetImage);
```

Comment:
not critical, a pure matter of style: I don't like actual logic in `catch` phrases. I would rather add a `return` in the end of the `try` and leave the `catch` empty, like:
```
try {
    await docker.inspectImage(targetImage);
    return;
} catch (err) { }

await docker.pull(targetImage);
```

Reply:
and also, if you don't use err so don't define it :)

Reply:
👍 
About the not declaring, is that possible? I'm getting an error if I try to.

Reply:
Just put a comment in `catch` block. Smth like `// do nothing`